File: ISEFront/api/ViewModels/CertificateGenerationDetailsViewModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ISEFront.api.ViewModels
{
    public class CertificateGenerationDetailsViewModel
    {
        [JsonProperty("subjectName")]
        public string SubjectName { get; set; }
        [JsonProperty("issuerName")]
        public string IssuerName { get; set; }
        [JsonProperty("notBefore")]
        public DateTimeOffset NotBefore { get; set; }
        [JsonProperty("notAfter")]
        public DateTimeOffset NotAfter { get; set; }
        [JsonProperty("privateKeyPassword")]
        public string PrivateKeyPassword { get; set; }
        [JsonProperty("keyStrength")]
        public int KeyStrength { get; set; }
        [JsonProperty("hashAlgorithm")]
        public string HashAlgorithm { get; set; }
    }
}

File: ISEFront/js/models/Dashboard/CertificateGenerationDetailsViewModel.ts
/// <reference path='../../_all.ts' />

module dashboard {
    'use strict';

    export class CertificateGenerationDetailsViewModel {
        constructor(
            public subjectName: string = "",
            public issuerName: string = "",
            public notBefore: Date = new Date(),
            public notAfter: Date = moment().add('years', 2).toDate(),
            public privateKeyPassword: string = '',
            public keyStrength: number = 2048,
            public hashAlgorithm: string = 'SHA512withRSA'
        ) { }
    }
}
